### PR TITLE
[feat-4937]: Update npm and node

### PR DESCRIPTION
**Pull request by OscarBakker**
**context**
We needed to upgrade the node version from 14 which dates from 2021 to 18 (we could consider 19 as well but 18 has lts status at the moment). 
For updating node I removed pro4jcompile and the prebuild node script. They seem unused. Also installed some linux packages needed in installing canvas from source: "canvas --build-from-source".

Ticket: [SIG-4937](https://gemeente-amsterdam.atlassian.net/browse/SIG-4937)

## Signalen

Before opening a pull request, please ensure:

- Make sure your PR title follows naming conventions: [feat-1234]: name feature
- Double-check your branch is based on `main` and targets `main`
- Pull request has tests (we are going for 100% coverage!)
- Code is well-commented, linted and follows project conventions
- Committed source code is headed by the correct SPDX license expression

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

[SIG-4937]: https://datapunt.atlassian.net/browse/SIG-4937?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,7 +10,11 @@ WORKDIR /app
 RUN apt-get update && apt-get install -y \
   git \
   netcat \
-  && rm -rf /var/lib/apt/lists/*
+  libcairo2-dev \
+  libpango1.0-dev \
+  libjpeg-dev \
+  libgif-dev \
+  librsvg2-dev   && rm -rf /var/lib/apt/lists/*
```
Comment (OscarBakker) [inline Dockerfile:17]: These are the packages needed by canvas. They were first part of node:14-stretch.

File: Dockerfile [hunk 2]
```diff
@@ -28,6 +32,8 @@ COPY .eslintrc.js \
   app.amsterdam.json \
   /app/
 
+RUN npm install canvas@2.11.0 --build-from-source
```
Comment (OscarBakker) [inline Dockerfile:35]: install from source using the above packages.

**Review by vdegraaf — COMMENTED** [review]
Nice! Interested how you got it working. 

One comment, in the CI there are some references to node 16, shouldn't they be updated/removed?

Screenshot 2023-01-27 at 13 46 56

**Comment by vdegraaf** [comment]
Is it possible that this update slows down the unit tests? Running them took much longer than usual. 
Node 14:
Screenshot 2023-01-27 at 13 55 32

Node 18: 
Screenshot 2023-01-27 at 13 45 34

**Comment by OscarBakker** [comment]
Updated GitHub testing hooks node version. Also updated jest which resulted in a small performance gain.

**Comment by OscarBakker** [comment]
> Is it possible that this update slows down the unit tests? Running them took much longer than usual. Node 14: Screenshot 2023-01-27 at 13 55 32
> 
> Node 18: Screenshot 2023-01-27 at 13 45 34

wordt waarschijnlijk sneller na deze, even apart getrokken want behoorlijk wat werk nog, zie ticket

https://gemeente-amsterdam.atlassian.net/browse/SIG-4991?atlOrigin=eyJpIjoiYjQxMTM5NmM1NDVjNGFmNDk1MzY5MjIzMTViNjc5ZTQiLCJwIjoiaiJ9

**Review by vdegraaf — APPROVED** [review]

**Pull request merged**